### PR TITLE
Branch duke command exception refactor

### PR DESCRIPTION
**Pull request by ngdeqi**
Related to #126 

Reduce coupling between pair unpair parsing classes with clientList, propertyList and pairingList classes.

Add more command pair unpair and check-specific exceptions

Add documentation for exceptions and commands

### CODE REVIEW

File: src/main/java/seedu/duke/command/Command.java [hunk 1]
```diff
@@ -16,9 +17,12 @@ public abstract class Command {
...
      * @param ui User interface object that handles interactions between user and the app.
      * @param storage Storage object that is responsible for storing and loading the app data.
-     * @param propertyList PropertyList object that handles all app interactions with the list clients.
-     * @param pairingList PairingList object that handles all interactions with the list clients paired with properties.
+     * @param propertyList PropertyList object that handles all app interactions with the list of properties.
+     * @param clientList ClientList object that handles all app interactions with the list of clients.
+     * @param pairingList PairingList object that handles all app interactions with the clients' rental of property.
+     * @throws DukeException When an irregularity occurs in the interaction with the data entities mentioned above.
      */
+
```
Comment (wilsonngja) [inline src/main/java/seedu/duke/command/Command.java:25]: Good job in the documentation. It seems to be adhering to the standards

File: src/main/java/seedu/duke/command/check/CommandCheckClient.java [hunk 2]
```diff
@@ -1,4 +1,4 @@
-package seedu.duke.command;
+package seedu.duke.command.check;
```
Comment (wilsonngja) [inline src/main/java/seedu/duke/command/check/CommandCheckClient.java:1]: Great effort in decoupling the exception.

File: src/main/java/seedu/duke/exception/check/checkproperty/CheckPropertyMissingFlagException.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+//@@author ngdeqi
+import seedu.duke.exception.check.ParseCheckException;
+
+import static seedu.duke.CommandStructure.CHECK_PROPERTY_FLAGS;
+import static seedu.duke.Messages.MESSAGE_CHECK_PROPERTY_WRONG_FORMAT;
+import static seedu.duke.Messages.MESSAGE_MISSING_FLAG;
+
+/**
+ * Representation of an exception where the user omits the property flag for the check property command.
+ */
```
Comment (wilsonngja) [inline src/main/java/seedu/duke/exception/check/checkproperty/CheckPropertyMissingFlagException.java:12]: You might want to rephrase this sentence to fits the standard indicated from the website.

File: src/main/java/seedu/duke/parsermanager/ParseCheckProperty.java [hunk 4]
```diff
@@ -1,93 +1,70 @@
...
-        if (!hasCorrectOrder) {
-            throw new IncorrectFlagOrderException(EXCEPTION);
+    private ArrayList<Integer> convertCheckCommandDetailsToInteger(ArrayList<String> checkDetailsString)
+            throws CheckNotIntegerException {
+        ArrayList<Integer> integerDetails = new ArrayList<>();
+        for (String detail : checkDetailsString) {
+            int integer;
+            try {
+                integer = Integer.parseInt(detail);
+                // Convert to 0-index
```
Comment (wilsonngja) [inline src/main/java/seedu/duke/parsermanager/ParseCheckProperty.java:61]: Good place to put a comment to help reader understand better.

File: src/main/java/seedu/duke/parsermanager/ParserManager.java [hunk 5]
```diff
@@ -98,9 +100,10 @@ public Parser parseCommand(String input) throws DukeParseException, ExistingPair
             if (isClient) {
                 return new ParseCheckClient(commandDetail, clientList);
             } else if (isProperty) {
-                return new ParseCheckProperty(commandDetail, propertyList);
+                return new ParseCheckProperty(commandDetail);
             } else {
-                throw new UndefinedSubCommandTypeException(MESSAGE_MISSING_SUB_COMMAND_TYPE);
+                throw new UndefinedSubCommandTypeException(MESSAGE_CHECK_CLIENT_WRONG_FORMAT
+                        + MESSAGE_CHECK_PROPERTY_WRONG_FORMAT);
```
Comment (wilsonngja) [inline src/main/java/seedu/duke/parsermanager/ParserManager.java:106]: You can consider changing this into a line of variable and pass the parameter inside a as 1 variable.

**Review by wilsonngja — COMMENTED** [review]
Overall, it's a job well done. I can see the efforts to adhere to the standards and the documentations were great too. LGTM

**Review by FeliciaBeatrice — APPROVED** [review]
everything looks good! LGTM

**Pull request merged**